core/presentation/ui/src/androidTest/java/com/pronoidsoftware/core/presentation/ui/DateTimePresentationUtilTest.kt

```diff
@@ -0,0 +1,190 @@
+package com.pronoidsoftware.core.presentation.ui
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.pronoidsoftware.core.domain.util.today
+import com.pronoidsoftware.testutil.jvmtest.core.data.time.TestClock
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.datetime.Clock
+import kotlinx.datetime.DateTimeUnit
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.format.DateTimeFormat
+import kotlinx.datetime.format.MonthNames
+import kotlinx.datetime.format.Padding
+import kotlinx.datetime.format.char
+import kotlinx.datetime.minus
+import kotlinx.datetime.plus
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
+import org.junit.Before
+import org.junit.Test
+
+class DateTimePresentationUtilTest {
```

Comment:
This is a type of test I would probably leave and rather make a UI test cover this. It's quite hard to put this reasoning in a strict rule, but I feel like when a normal unit test already requires instrumentation we can also involve the UI right away to cover the whole functionality. 
My feeling is that even though the tests here are correct and test relevant logic, they probably won't ever catch a real bug, since the logic is not that error-prone as an isolated piece (it's unlikely it will ever change, it's mostly handled by the Kotlinx DateTime API and if it changes it's not very hard to change it to a different formatting or so).

Experience has shown that there are these types of tests that never break, since the code is just not very likely to change. So to summarize, I would somehow cover this functionality with an integration/UI test, but probably not make a whole dedicated test class for this. There are probably better uses for your time.

(Of course now that you have it, keep it 😄 )